File: 8kyu/is-there-a-vowel-in-there/is-there-a-vowel-in-there.py
from typing import List, Union

# def is_vow(inp: list[int]) -> list[Union[int, str]]:
#   res=[]
#   vowels = ['a', 'e', 'i', 'o', 'u']
#   for num in inp:
#     res.append(chr(num)) if chr(num) in vowels else res.append(num)
#   return res

def is_vow(inp: list[int]) -> list[Union[int, str]]:
  return [chr(num) if chr(num) in 'aeiou' else num for num in inp]

# print(is_vow(117))
# print(is_vow([118,117,120]))
print(is_vow([118,117,120,121,117,98,122,97,120,106,104,116,113,114,113,120,106]))

File: 8kyu/is-there-a-vowel-in-there/is-there-a-vowel-in-there.js
/* 
PREP
p - array
r - return the array with vowels in letters 
e - [101,121,110,113,113,103,121,121,101,107,103] => ["e",121,110,113,113,103,121,121,"e",107,103];
p
- loop through the array 
- if vowel, change it to letter
*/

function isVow(a) {
	const getString = (code) => String.fromCharCode(code);
	const vowel = /[aeiou]/;
	return a.map((code) =>
		vowel.test(getString(code)) ? getString(code) : code,
	);
}
